### DISCUSSION TITLE
General flow of how DiscordChatExporter works?

### DISCUSSION THREAD

**Discussion by eelegiap** [Tyrrrz/DiscordChatExporter#1109] [Q&A] [answered]
Hi, this is a great repository, thanks for your great work on it. I was wondering if there is any info about how DiscordChatExporter works in general? I don't really understand the backend of how this works after I give my user token to the CLI. It would be great to see a flow of information or a general overview of its framework - I know it must be complicated. Thanks in advance.

**Comment by Tyrrrz** [comment, marked as answer]
It's pretty simple, actually. Once you provide the token, it uses it to make authenticated requests to the Discord API, pulling the list of guilds (servers) you're part of, their channels and, when you choose to export one of them, iteratively pulls all the messages belonging to that channel. Additional data processing happens as well, such as resolving mentions, parsing markdown, etc. Then all of that is put through one of the exporters (depending on the format).
